sky/provision/provisioner.py

```diff
@@ -138,7 +140,14 @@ def bulk_provision(
     prev_cluster_ever_up: bool,
     log_dir: str,
 ) -> provision_common.ProvisionRecord:
-    """Provisions a cluster and wait until fully provisioned."""
+    """Provisions a cluster and wait until fully provisioned.
+
+    Raises:
+        StopFailoverError: If the cluster was ever up and the provisioning
+            process failed.
```

Comment:
Is it more accurate to say "Raised when during failover cleanup, tearing down any potentially live cluster failed despite retries"